new-client/src/plugins/Anchor/AnchorView.js

```diff
@@ -39,11 +39,11 @@ class AnchorView extends React.PureComponent {
     cleanUrl: false,
   };
 
-  async componentDidMount() {
-    // Set anchor URL initially
-    this.setState({
-      anchor: await this.props.model.getAnchor(),
-    });
+  componentDidMount() {
+    // We check if anchor is a string, to avoid error in render() below.
+    // This because the getAnchor() sometimes returns a Promise instead of a string.
+    typeof anchor === "string" &&
```

Comment:
`anchor` is not defined here and won't be of type string. When i test your implementation, the view results in an empty field: 
![image](https://github.com/hajkmap/Hajk/assets/43059093/b99996bd-7783-4b66-a373-f8a1ea4142e4)

(It renders an empty field if you have the anchor plugin enabled at start).


I think this issue could be resolved with:

```js
async componentDidMount() {
    const a = await this.props.model.getAnchor();
    this.setState({ anchor: a });
    // Subscribe to changes to anchor URL caused by other components. This ensure
    // that we have a live update of the anchor whether user does anything in the map.
    this.props.globalObserver.subscribe(
      "core.mapUpdated",
      ({ url, source }) => {
        this.setState({
          anchor: this.appendCleanModeIfActive(url),
        });
      }
    );
  }
```

Maybe you could test that and see if it works for you as well? The strange thing is that the original implementation also works for me...

Reply:
Interesting. I tried having the dela-window opened at start aswell, and the link does appear. But I'll try your fix :)

Reply:
So your codes seems to be working fine. Although, if I scroll while the page is starting the link does not appear.

Reply:
You mean it doesn't get updated?

The `componentDidMount` part runs only on initial component load. The following updates of anchor string are taken care of elsewhere, please ensure to use the same methodology (`await this.props.model.getAnchor()`) before calling `setState()`.

Reply:
Hmm... Seems like `getAnchor` never resolves in some cases... It's not pretty but this should do the trick: 

```js
async componentDidMount() {
    // Subscribe to changes to anchor URL caused by other components. This ensure
    // that we have a live update of the anchor whether user does anything in the map.
    this.props.globalObserver.subscribe(
      "core.mapUpdated",
      ({ url, source }) => {
        this.setState({
          anchor: this.appendCleanModeIfActive(url),
        });
      }
    );
    const a = await this.props.model.getAnchor();
    this.setState({ anchor: a });
  }
```

(We have to make sure the subscription is made. If `getAnchor` never resolves, the subscription is never made, and the anchor-url is never updated).

Reply:
This works on my end :)